development/extensions/skeleton_extension.rst

```diff
@@ -0,0 +1,750 @@
+this would be ``acme``. **This field is required.**
+
+Package name
+^^^^^^^^^^^^
+
+The package name is the name of the extension. Valid naming conventions must
+start with a letter and contain letters and numbers only. In the case of the 
+Acme Demo extension, this would be ``demo``. **This field is required.**
+
+Display name
+^^^^^^^^^^^^
+
+This is the name (or title) of your extension in plain English. This is what the user will
+see when your extension appears in the phpBB ACP Extension Manager. In the case of the 
+Acme Demo extension, this would be ``Acme Demo``. **This field is required.**
+
+Description
+^^^^^^^^^^^
+
+A brief description of your extension. This will be visible to users in the Details page
+of your extension in phpBB's ACP Extension Manager.
+
+Version
+^^^^^^^
+
+The version of your extension. phpBB's coding guidelines require an ``x.y.z`` semantic 
+versioning format, with optional ``-dev``, ``-alpha``, ``-beta`` and ``-RC`` pre-release suffixes.
+When left blank, the Skeleton Extension will use ``1.0.0-dev`` by default.
+
+Date of the extension
+^^^^^^^^^^^^^^^^^^^^^
+
+Date the extension was created. Dates must be in the ``YYYY-MM-DD`` format. 
+When left blank, the Skeleton Extension will use the current date by default.
+
+Homepage of the extension
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+An optional link to your extension's homepage. This can be, for example, an extension's
+GitHub repository or any other site related to your extension.
+
+Authors
+-------
+
+Here you can name your extension's author(s). Use the ``Add author`` button
+to add additional authors.
+
+Author name
+^^^^^^^^^^^
+
+The name of an extension's author. Must not be empty.
+
+Author email
+^^^^^^^^^^^^
+
+The email address of an extension's author. (Optional)
+
+Author homepage
+^^^^^^^^^^^^^^^
+
+The homepage URL of an extension's author. (Optional)
+
+Author role
+^^^^^^^^^^^
+
+The role of an extension's author. For example, lead developer, contributor, 
+translator, etc. (Optional)
+
+Requirements
+------------
+
+Requirements are where you define the minimum (and maximum) PHP and phpBB versions
+your extension needs to be able to perform.
+
+Minimum phpBB requirement of the extension
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Typically this should be set to the version of phpBB in which you are developing
+and testing your extension.
+
+The version should be defined using `Composer's version 
+constraints <https://getcomposer.org/doc/articles/versions.md>`_. 
+
+For example, to require phpBB 3.1.4 or greater, use ``>=3.1.4``.
+
+Maximum phpBB requirement of the extension
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To mitigate compatibility breaks, it is common to set the maximum version
+to the next major release version of phpBB.
+
+For example, if your extension is being developed for phpBB 3.1.x, you can
+set the maximum requirement to phpBB 3.2.x by using ``<3.2.0@dev``.
```

Comment:
I know. The reasoning for it was we could give warnings in phpBB but not physically prevent installation; as we don't use semver we make life shitty for extension authors in this regard.